### PR TITLE
added undersampling

### PR DESCRIPTION
**Pull request by SingarajuP**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,5 @@
-#Ignore Mac system files
-.DS_store
+.DS_Store
 
 #Ignore models in the folder
-https://drive.google.com/drive/folders/1jmSVF-PuWZd1W1kHmZ0mhea1H2iZADix?usp=sharing
\ No newline at end of file
+models/
+data/raw/
```
Comment (rfazeli) [inline .gitignore:5]: It's always a good idea to have a newline at the end of all your files.

**Review by rfazeli — CHANGES_REQUESTED** [review]
Hi Phani, 

Thanks for making these changes. Do you think you can push .py copies of your notebooks so I can review them?

Thanks

File: .gitignore (hunk 1 above)
Comment (SingarajuP) [reply .gitignore:9]: I have pushed .py files. 
Ok. I will take care of this from next time.

File: notebooks/movie_subtitles_results-undersampling.py [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+# In[5]:
+
+
+df.title.to_csv('../models/Titles.csv',index=False, header=False)
+
+
+# In[6]:
+
+
+df_sub=pd.read_csv("../data/raw/movies_subtitles.csv")
```
Comment (rfazeli) [inline notebooks/movie_subtitles_results-undersampling.py:40]: Can you include these files in the relevant folders? `Titles.csv` should also be under data/raw and maybe rename it to `movies_titles.csv` for consistency.

File: notebooks/movie_subtitles_results-undersampling.py [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+import pandas as pd
+import numpy as np
+import re
+from string import punctuation 
+
+
+# In[2]:
+
+
+df=pd.read_csv("../data/raw/movies_meta.csv")
```
Comment (rfazeli) [inline notebooks/movie_subtitles_results-undersampling.py:16]: Add this to your data folder as well

**Review by rfazeli — APPROVED** [review]
This looks great, thanks Phani! Just added some small comments. It'd be good to give your branches more descriptive names too e.g. movie_data

**Comment by rfazeli** [comment]
It's also a good idea to remove this branch once you merge it to avoid unused branches lying around.

File: notebooks/combining_emotions_neutral_models.py [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+from nltk.stem import WordNetLemmatizer
+from nltk.tokenize import sent_tokenize,word_tokenize
+from nltk.corpus import stopwords
+
+
+# In[4]:
+
+
+df_6emotions = pd.read_pickle('../data/raw/merged_training.pkl')
+df_senti=pd.read_csv("../data/raw/train.csv", encoding= 'unicode_escape')
```
[the remark this answers is not in the document]
Comment (SingarajuP) [reply notebooks/combining_emotions_neutral_models.py:28]: Hi Reza, I will change the names of those files. 
Yes, I had checked now for the data/raw directory in the repository. But surprisingly it is not there. I have it in my local. What should I do for it to appear in repository also?

File: notebooks/combining_emotions_neutral_models.py [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+
+#Logistic Regression with multinomial
+lr_mn = LogisticRegression(multi_class='multinomial', solver='lbfgs')
+lr_mn.fit(X_under, y_under)
+
+
+# In[40]:
+
+
+ypred_lr_mn=lr_mn.predict(Xtest_tfidf)
```
[the remark this answers is not in the document]
Comment (SingarajuP) [reply notebooks/combining_emotions_neutral_models.py:225]: Yes. I will work on it.

File: notebooks/exploratory_analysis.py [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+
+df['text'][0]
+
+
+# In[9]:
+
+
+#Getting the count of words in each row. 
+df['Number_words']=df['text'].str.split().apply(len)
```
[the remark this answers is not in the document]
Comment (SingarajuP) [reply notebooks/exploratory_analysis.py:72]: Oh Ok.

File: notebooks/movie_subtitles_results-undersampling.py (hunk 2 above)
Comment (SingarajuP) [reply notebooks/movie_subtitles_results-undersampling.py:40]: Yes. I have them in the raw folder which I put in .gitignore

Comment (SingarajuP) [reply notebooks/movie_subtitles_results-undersampling.py:40]: The Titles.csv is the one (which I will remove later) I made to see what are the titles of the movies available.

**Comment by rfazeli** [comment]
Thanks very muc for making the changes, Phani! Let's go ahead and merge this PR.

Also it looks like your test accuracy has slightly improved after balancing the test dataset. At least for the logistic regression model. Right?

**Pull request merged**